src/java.base/share/classes/java/net/URLDecoder.java

```diff
@@ -89,7 +91,11 @@ public class URLDecoder {
     private URLDecoder() {}
 
     // The default charset
-    static String dfltEncName = URLEncoder.dfltEncName;
+    private static final String DEFAULT_ENCODING_NAME;
+
+    static {
+        DEFAULT_ENCODING_NAME = StaticProperty.fileEncoding();
+    }
 
```

Comment:
Can we just in-line the use of `StaticProperty.fileEncoding()` to the call site as `StaicProperty.FILE_ENCODING` is already stored in a static final field

```suggestion
```

https://github.com/openjdk/jdk/blob/77a50105f0c4a4cb7286dda13c633f1e69295210/src/java.base/share/classes/jdk/internal/util/StaticProperty.java#L54-L74